Document/0x06h-Testing-Platform-Interaction.md

```diff
@@ -3,365 +3,2753 @@
+```
+$ r2 -qc 'izz~PropertyList' ./Telegram\ X
+
+0x0015d2a4 ascii <?xml version="1.0" encoding="UTF-8" standalone="yes"?>\n<!DOCTYPE plist PUBLIC 
+"-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">\n<plist version="1.0">
+...<key>com.apple.security.application-groups</key>\n\t\t<array>
+\n\t\t\t<string>group.ph.telegra.Telegraph</string>...
+
+0x0016427d ascii H<?xml version="1.0" encoding="UTF-8"?>\n<!DOCTYPE plist PUBLIC 
+"-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">\n<plist version="1.0">\n
+<dict>\n\t<key>cdhashes</key>...
+```
+
+In both cases (Binwalk or Radare2) we were able to extract the same two `plist` files. If we inspect the first one (0x0015d2a4) we see that we were able to completely recover the [original entitlements file from Telegram](https://github.com/peter-iakovlev/Telegram-iOS/blob/77ee5c4dabdd6eb5f1e2ff76219edf7e18b45c00/Telegram-iOS/Telegram-iOS-AppStoreLLC.entitlements "Telegram-iOS-AppStoreLLC.entitlements original file").
+
+> Note: the `strings` command will not help here as it will not be able to find this information. Better use grep with the `-a` flag directly on the binary or use Radare2 (`izz`)/Rabin2 (`-zz`).
+
+If you access the app binary on the jailbroken device (e.g via SSH), you can use grep with the `-a, --text` flag (treats all files as ASCII text):
+
+```bash
+$ grep -a -A 5 'PropertyList' /var/containers/Bundle/Application/
+    15E6A58F-1CA7-44A4-A9E0-6CA85B65FA35/Telegram X.app/Telegram\ X
+
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+	<dict>
+		<key>com.apple.security.application-groups</key>
+		<array>
+        ...
+```
+
+Play with the `-A num, --after-context=num` flag to display more or less lines. You may use tools like the ones we presented above as well, if you have them also installed on your jailbroken iOS device.
+
+> This method should work even if the app binary is still encrypted (it was tested against several App Store apps).
+
+##### Source Code Inspection
+
+After having checked the `<appname>.entitlements` file and the `Info.plist` file, it is time to verify how the requested permissions and assigned capabilities are put to use. For this, a source code review should be enough.
+
+Pay attention to:
+
+- whether the *purpose strings* in the `Info.plist` file match the programmatic implementations.
+- whether the capabilities registered are used in such a way that no confidential information is leaking.
+
+#### Dynamic Analysis
+
+Use static analysis first to take note of the included permissions included in the relevant files. After that, go through the application and check whether the application communicates with other applications or with back ends. Check whether the information retrieved using the permissions and capabilities is used for malicious purposes or the permissions and capabilities themselves are over-asked/under-utilized.
```

Comment:
> included permissions included in the relevant files.

To

> included permissions in the relevant files

Reply:
done, the paragraph was completely modified